src/Http/Import/ImportTermRequestBodyParser.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Http\Import;
+
+use CultuurNet\UDB3\Http\Request\Body\RequestBodyParser;
+use CultuurNet\UDB3\Model\Import\Taxonomy\Category\CategoryResolverInterface;
+use CultuurNet\UDB3\Model\ValueObject\Taxonomy\Category\CategoryID;
+use Psr\Http\Message\ServerRequestInterface;
+use stdClass;
+
+final class ImportTermRequestBodyParser implements RequestBodyParser
+{
+    private CategoryResolverInterface $placeCategoryResolver;
+
+    public function __construct(CategoryResolverInterface $placeCategoryResolver)
+    {
+        $this->placeCategoryResolver = $placeCategoryResolver;
+    }
+
+    public function parse(ServerRequestInterface $request): ServerRequestInterface
+    {
+        $json = $request->getParsedBody();
+
+        // Attempt to add label and/or domain to terms, or fix them if they're incorrect.
+        if (isset($json->terms) && is_array($json->terms)) {
+            $json->terms = array_map(
+                function (stdClass $term) {
+                    if (isset($term->id) && is_string($term->id)) {
+                        $id = $term->id;
+                        $category = $this->placeCategoryResolver->byId(new CategoryID($id));
+
+                        if ($category) {
+                            $term->{'label'} = $category->getLabel()->toString();
+                            $term->{'domain'} = $category->getDomain()->toString();
```

Comment:
I think you can just use `$term->label` and `$term->domain` here, no need for the curly braces and quotation marks.

Reply:
Indeed the curly brackets are not needed. I had the `$term` variable first typed as `object` when doing that PHPStan gives following errors:
```
 ------ -------------------------------------------------- 
  Line   src/Http/Import/ImportTermRequestBodyParser.php   
 ------ -------------------------------------------------- 
  35     Access to an undefined property object::$label.   
  36     Access to an undefined property object::$domain.  
 ------ -------------------------------------------------- 
 ```

When changing the type to stdClass those errors are gone.